keploy.sh

```diff
@@ -75,6 +75,100 @@ installKeploy (){
 
     ARCH=$(uname -m)
 
+    # Renders a text based list of options that can be selected by the
+    # user using up, down and enter keys and returns the chosen option.
+    #
+    #   Arguments   : list of options, maximum of 256
+    #                 "opt1" "opt2" ...
+    #   Return value: selected index (0 for opt1, 1 for opt2 ...)
+    select_option() {
+        ESC=$(printf "\033")
+
+        cursor_blink_on() {
+            printf "$ESC[?25h"
+        }
+
+        cursor_blink_off() {
+            printf "$ESC[?25l"
+        }
+
+        cursor_to() {
+            printf "$ESC[$1;${2:-1}H"
+        }
+
+        print_option() {
+            printf "   $1 "
+        }
+
+        print_selected() {
+            printf "  $ESC[7m $1 $ESC[27m"
+        }
+
+        get_cursor_row() {
+            IFS=';' read -sdR -p $'\E[6n' ROW COL
+            echo ${ROW#*[}
+        }
+
+        key_input() {
+            read -s -n3 key 2>/dev/null >&2
+            if [ "$key" = "$ESC[A" ]; then
+                echo up
+            elif [ "$key" = "$ESC[B" ]; then
+                echo down
+            elif [ "$key" = "" ]; then
+                echo enter
+            fi
+        }
+
+        options="$1"
+        set -- $options  # Set positional parameters
+        for opt do
+            printf "\n"
+        done
+
+        local lastrow
+        lastrow=$(get_cursor_row)
+        local startrow
+        startrow=$((lastrow - $#))
+
+        trap "cursor_blink_on; stty echo; printf '\n'; exit" 2
+        cursor_blink_off
+
+        local selected=0
+        while true; do
+            local idx=0
+            for opt; do
+                cursor_to $((startrow + idx))
+                if [ "$idx" -eq "$selected" ]; then
+                    print_selected "$opt"
+                else
+                    print_option "$opt"
+                fi
+                idx=$((idx + 1))
+            done
+
+            case $(key_input) in
+                enter) break;;
+                up) selected=$((selected - 1))
+                    if [ "$selected" -lt 0 ]; then
+                        selected=$(( $# - 1 ))
+                    fi;;
+                down) selected=$((selected + 1))
+                    if [ "$selected" -ge $# ]; then
+                        selected=0
+                    fi;;
+            esac
+        done
+
+        cursor_to "$lastrow"
+        printf "\n"
+        cursor_blink_on
+
+        return "$selected"
+    }
+
+    ARCH=$(uname -m)
```

Comment:
Why is this redeclared in this block?

Reply:
@shivamsouravjha sorry for that. Updated the code!